### PR TITLE
[BEAM-1338] Moves ThreadPool creation to a util function

### PR DESCRIPTION
**Pull request by chamikaramj**
Moves ThreadPool creation to a util function.
Records and resets logging level due to this being reset by  apitools when used with a ThreadPool.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @aaltay

File: sdks/python/apache_beam/internal/util.py [hunk 1]
```diff
@@ -92,3 +97,31 @@ def insert_values_in_args(args, kwargs, values):
...
+
+  # ThreadPool crashes in old versions of Python (< 2.7.5) if created
+  # from a child thread. (http://bugs.python.org/issue10015)
+  if not hasattr(threading.current_thread(), '_children'):
+    threading.current_thread()._children = weakref.WeakKeyDictionary()
+  pool = ThreadPool(min(pool_size, len(inputs)))
+  try:
+    # We record and reset logging level here since 'apitools' library Beam
+    # depends on updates the logging level when used with a threadpool.
+    # TODO: Remove this once above issue in 'apitools' is fixed and when it is
```
Comment (aaltay) [inline sdks/python/apache_beam/internal/util.py]: When should we revisit this TODO? Is there an apitools issue we can link to?

Comment (chamikaramj) [inline sdks/python/apache_beam/internal/util.py]: Added the Github issue to comments.

**Comment by chamikaramj** [comment]
Thanks. PTAL.

**Comment by aaltay** [comment]
Thank you @chamikaramj LGTM.

**Comment by chamikaramj** [comment]
Thanks.

R: @robertwb can you merge ?

**Comment by davorbonaci** [comment]
Merging.

**Comment by chamikaramj** [comment]
Thanks Davor.

**Pull request closed without merging**